audb/core/lock.py

```diff
@@ -0,0 +1,61 @@
+import types
+import typing
+
+import filelock
+
+import audeer
+
+import audb.core.define as define
+
+
+class FolderLock:
+
+    def __init__(
+            self,
+            folders: typing.Union[str, typing.Sequence[str]],
+            *,
+            timeout: float = -1,
+    ):
+        r"""Lock one or more folders.
+
+        Waits until the locks of all folders can be acquired.
+        While a folder is locked,
+        a file '.lock' will be created inside the folder.
+
+        Args:
+            folders: path to one or more folders that should be locked
+            timeout: maximum wait time if another thread or process is already
+                accessing one or more locks. If timeout is reached,
+                ``None`` is returned. If timeout < 0 the method will block
+                until the resource can be accessed
```

Comment:
```suggestion
            timeout: maximum wait time if another thread or process
                is already accessing one or more locks.
                If timeout is reached, ``None`` is returned.
                If timeout < 0 the method will block
                until the resource can be accessed
```

Reply:
> How does this lock multiple fodlers?

We have tests for it.

Reply:
Actually we do not return `None` but raise an error, I rephrased to:

```
        r"""Lock one or more folders.

        Waits until the locks of all folders can be acquired.
        While a folder is locked,
        a file '.lock' will be created inside the folder.

        Args:
            folders: path to one or more folders that should be locked
            timeout: maximum wait time if another thread or process
                is already accessing one or more locks.
                If timeout is reached,
                an exception is raised.
                If timeout < 0 the method will block
                until the resource can be accessed

        Raises:
            :class:`filelock.Timeout`: if a timeout is reached

        """
```